projects/rocprofiler-sdk/source/lib/output/generateRocpd.cpp

```
@@ -1399,6 +1628,219 @@ write_rocpd(
+                    .workgroup_id    = {.x = record.workgroup_id.x,
+                                     .y = record.workgroup_id.y,
+                                     .z = record.workgroup_id.z},
+                    .dispatch_id     = record.dispatch_id,
+                    .wave_issued     = std::nullopt,
+                    .inst_type       = std::nullopt,
+                    .wave_count      = std::nullopt,
+                    .flags           = std::nullopt,
+                    .snapshot        = std::nullopt,
+                    .memory_counters = std::nullopt};
+
+                // Add stochastic-specific data only for stochastic records
+                if(sampling_method == "stochastic")
+                {
+                    if constexpr(std::is_same_v<
+                                     std::decay_t<decltype(pc_sampling_gen)>,
+                                     generator<rocprofiler_tool_pc_sampling_stochastic_record_t>>)
+                    {
+                        const auto& stochastic_record = record;
+
+                        const char* inst_type_name =
+                            rocprofiler_get_pc_sampling_instruction_type_name(
+                                static_cast<rocprofiler_pc_sampling_instruction_type_t>(
+                                    stochastic_record.inst_type));
+
+                        const char* reason_not_issued_name =
+                            rocprofiler_get_pc_sampling_instruction_not_issued_reason_name(
+                                static_cast<
+                                    rocprofiler_pc_sampling_instruction_not_issued_reason_t>(
+                                    stochastic_record.snapshot.reason_not_issued));
+
+                        std::string inst_type_str =
+                            inst_type_name ? std::string(inst_type_name)
+                                           : std::to_string(stochastic_record.inst_type);
```

So, if the `inst_type_name  == nullptr`, we're going to serialize the enum value? Should we report any error message as this should not happen?